lib/interactor/context.rb

```diff
@@ -121,10 +121,22 @@ def failure?
     #
     # Raises Interactor::Failure initialized with the Interactor::Context.
     def fail!(context = {})
-      error_context = Context.new
-      context.each { |key, value| error_context[key.to_sym] = value }
-      error_context.failure = @failure = true
-      raise Failure, error_context
+      context.each { |key, value| self[key.to_sym] = value }
+
+      redaction_message = 'REDACTED'
+      self.each_pair do |name, value|
+        if value.is_a?(Hash)
+          if value.has_key?(:password)
+            self[name.to_sym][:password] = redaction_message
+          end
+          if value.has_key?(:password_confirmation)
+            self[name.to_sym][:password_confirmation] = redaction_message
+          end
```

Comment:
If you add another key to redact in the future, it'll be a new `if` statement here. Probably worth DRY-ing it up to just read from a list of redacted keys

```ruby
REDACTED_FIELDS = %i[password password_confirmation]

# ...

REDACTED_FIELDS.each do |field|
  self[name][field] = REDACTION_MESSAGE if value.key?(field)
end
```